### DISCUSSION TITLE
A simple set operation takes 200-300ms（zustand V3）

### DISCUSSION THREAD

**Discussion by LZY622** [pmndrs/zustand#2019] [Bug report]
![image](https://github.com/pmndrs/zustand/assets/42869166/8f6d1f01-1483-46f2-8be4-7c2fba108a57)
The content enclosed in the red box is the time consumption of a simple set operation.

After upgrading the version, the time-consuming is shortened by 50%.

some information：
1. There are 86 keys in the store
2. There are multiple stores in the project
3. The project is complex and difficult to abstract into a demo，This bug cannot be reproduced in the demo，so sorry~~~

Please give me some suggestions and optimization ideas.Thanks~~~~

**Comment by dai-shi** [comment]
It would be probably hard to get help from someone without a reproduction.

**Comment by dbritto-dev** [comment]
@LZY622 you could try using Map instead of plain object

**Reply by LZY622** [reply]
Okay, thank you very much. I'll give it a try
